lib/async.js

```diff
@@ -449,35 +455,40 @@
     async.detect = doParallel(_detect);
     async.detectSeries = doSeries(_detect);
 
+    function _createTester(eachfn, check, defaultValue) {
+        return function(arr, limit, iterator, cb) {
+            function done() {
+                if (cb) cb(defaultValue);
```

Comment:
This is only done at the last step, `noop` actually makes it more complicated because then at the other step (iterator) it has to check `if (cb === noop)`